core/includes/theme.inc

```diff
@@ -1927,12 +1927,35 @@ function theme_dropbutton_wrapper($variables) {
 }
 
 /**
+ * Preprocess variables for theme_image().
+ *
+ * @since 1.8.0 Function added. Populates $variables['attributes'] before
+ *   theme_image() is called.
+ */
+function template_preprocess_image(&$variables) {
+  // Accept either URI or path key. But both are handled by file_create_url().
+  $path_or_uri = $variables['uri'] ? $variables['uri'] : $variables['path'];
```

Comment:
...newbie question: should we be checking the case when none of the variables exist here?